android/src/main/java/com/twiliorn/library/CustomTwilioVideoView.java

```diff
@@ -487,6 +499,38 @@ public void connectToRoom() {
             connectOptionsBuilder.dataTracks(Collections.singletonList(localDataTrack));
         }
 
+        // H264 Codec Support Detection: https://www.twilio.com/docs/video/managing-codecs
+        HardwareVideoEncoderFactory hardwareVideoEncoderFactory = new HardwareVideoEncoderFactory(null, true, true);
+        HardwareVideoDecoderFactory hardwareVideoDecoderFactory = new HardwareVideoDecoderFactory(null);
+
+        boolean h264EncoderSupported = false;
+        for (VideoCodecInfo videoCodecInfo : hardwareVideoEncoderFactory.getSupportedCodecs()) {
+            if (videoCodecInfo.name.equalsIgnoreCase("h264")) {
+                h264EncoderSupported = true;
+                break;
+            }
+        }
+        boolean h264DecoderSupported = false;
+        for (VideoCodecInfo videoCodecInfo : hardwareVideoDecoderFactory.getSupportedCodecs()) {
+            if (videoCodecInfo.name.equalsIgnoreCase("h264")) {
+                h264DecoderSupported = true;
+                break;
+            }
+        }
+
+        boolean isH264Supported = h264EncoderSupported && h264DecoderSupported;
+
+        Log.d("RNTwilioVideo", "isH264Supported: " + isH264Supported);
+
+        VideoCodec videoCodec = isH264Supported && this.enableH264Codec ? (new H264Codec()) : (new Vp8Codec());
+
+        WritableMap event = new WritableNativeMap();
+        event.putBoolean("isH264Supported", isH264Supported);
```

Comment:
In order to future-proof this event, ,maybe it would make more sense to have the event return the list of supported codecs.

Reply:
Let me update my PR to reflect this. Thanks!